### PR TITLE
Feature/covid

### PR DESCRIPTION
**Pull request by getjump**
#101

### CODE REVIEW

File: bot/filters.py [hunk 1]
```diff
@@ -22,6 +26,20 @@ def filter(self, message) -> bool:
...
+
+    def filter(self, message: Message) -> bool:
+        if get_debug():
+            return True
+        if message.reply_to_message != None:
+            return message.from_user.id in {
+                a.user.id for a in message.chat.get_administrators()
+            }
+        else:
+            return True
```
Comment (C-Pro) [inline bot/filters.py]: Maybe False?

File: bot/skills/covid_mode.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+                "quarantined_until": until
+            }
+        })
+
+    def get_quarantined(self):
+        self._coll.find({ "quarantined_until": { "$lte": datetime.now() } })
+
+    def dead(self, user_id: str, mute_min: int):
+        self._coll.update_one({"_id": user_id}, {
+            "$inc": {"shot_counter": 1, "dead_counter": 1, "total_time_in_club": mute_min * 60},
```
Comment (C-Pro) [inline bot/skills/covid_mode.py]: time_in_club is in seconds?

Comment (egregors) [reply bot/skills/covid_mode.py]: looks like unrelated copy|paste code from another mode 😐 😐 😐

File: bot/skills/covid_mode.py [hunk 3]
```diff
@@ -409,17 +420,44 @@ def catch_message(update: Update, context: CallbackContext):
...
+    sha1(img[-100:]).hexdigest()
+
+
+def is_avatar_has_mask(img: bytearray, context: CallbackContext) -> bool:
+    if (img is None) or len(img) < 100:
+        return False
+
+    # lookup existing value in cache
+    hash = hashImg(img)
+    isGood = context.chat_data['avatar_mask_cache'].get(hash)
```
Comment (egregors) [inline bot/skills/covid_mode.py]: @C-Pro this is not Go!

Comment (C-Pro) [reply bot/skills/covid_mode.py]: What's wrong?

Comment (C-Pro) [reply bot/skills/covid_mode.py]: Oh camelCase :))))
I officially don't give a *uck

Comment (C-Pro) [reply bot/skills/covid_mode.py]: @getjump https://github.com/getjump/vldc-bot/pull/2

File: bot/skills/covid_mode.py [hunk 4]
```diff
@@ -110,6 +112,16 @@ def is_user_infected(self, user_id: str):
...
         }) is not None
 
+    def add_lethality(self, user_id: str, since: datetime):
+        self._coll.update_one({"_id": user_id}, {
+            "$set": {
+                "lethaled_since": since
+            }
+        })
+
+    def get_lethaled(self):
```
Comment (C-Pro) [inline bot/skills/covid_mode.py]: I think we'd better use the term "deceased".

File: bot/skills/covid_mode.py [hunk 5]
```diff
@@ -224,6 +236,15 @@ def stop(update: Update, context: CallbackContext):
                                                  can_send_media_messages=True,
                                                  can_send_other_messages=True,
                                                  can_send_messages=True)
+        lethaled = _db.get_lethaled()
+
+        if lethaled is not None:
+            for user in lethaled:
+                context.bot.restrict_chat_member(get_group_chat_id(), user["_id"],
+                                                 can_add_web_page_previews=True,
```
Comment (C-Pro) [inline bot/skills/covid_mode.py]: Maybe false?

File: bot/skills/covid_mode.py [hunk 6]
```diff
@@ -341,17 +362,27 @@ def random_cough(bot: Bot, queue: JobQueue):
...
+            days_count = (datetime.now() - _user['infected_since']).days
+            if _rng <= 1 / math.exp(1 / (LETHALITY_RATE * days_count)):
+                chance = .0
+                bot.restrict_chat_member(get_group_chat_id(), _user['id'],
+                                         can_add_web_page_previews=False,
+                                         can_send_media_messages=False,
+                                         can_send_other_messages=False,
+                                         can_send_messages=False)
+                message = message + \
+                (f"{full_name} умер от коронавируса F") + "\n"
```
Comment (C-Pro) [inline bot/skills/covid_mode.py]: ```{full_name} was lost due to coronavirus outbreak```

**Pull request merged**